src/nvim/api/ui.c

```diff
@@ -24,21 +23,13 @@ typedef struct {
 static PMap(uint64_t) *connected_uis = NULL;
 
 void remote_ui_init(void)
+    FUNC_ATTR_NOEXPORT
 {
   connected_uis = pmap_new(uint64_t)();
-  // Add handler for "attach_ui"
-  String method = cstr_as_string("ui_attach");
-  MsgpackRpcRequestHandler handler = {.fn = remote_ui_attach, .async = false};
-  msgpack_rpc_add_method_handler(method, handler);
-  method = cstr_as_string("ui_detach");
-  handler.fn = remote_ui_detach;
-  msgpack_rpc_add_method_handler(method, handler);
-  method = cstr_as_string("ui_try_resize");
-  handler.fn = remote_ui_try_resize;
-  msgpack_rpc_add_method_handler(method, handler);
 }
 
 void remote_ui_disconnect(uint64_t channel_id)
+    FUNC_ATTR_NOEXPORT
```

Comment:
@justinmk functions already "opt in" by being placed in "api/*.c" (and being non-static). One could remove these by making `connected_uis` global, but I think it would be reasonable to have non-static initializers inside api modules without needing to mark every other function `FUNC_ATTR_API`